vm/src/hint_processor/builtin_hint_processor/bootloader/execute_task_hints.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+
+use crate::hint_processor::builtin_hint_processor::bootloader::vars;
+use crate::hint_processor::builtin_hint_processor::hint_utils::insert_value_from_var_name;
+use crate::hint_processor::hint_processor_definition::HintReference;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::exec_scope::ExecutionScopes;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::vm_core::VirtualMachine;
+
+/// Implements %{ ids.program_data_ptr = program_data_base = segments.add() %}.
```

Comment:
It seems weird that we assign to `ids_data` and `exec_scopes`...

Reply:
Yeah, I guess these variables have different usages? Anyway, that's what the hint does, and the will of the hint is sacred.